azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java

```diff
@@ -411,27 +415,47 @@ private void launchComposite(final Context context,
 
         UUID groupId = null;
         String meetingLink = null;
+        /* <ROOMS_SUPPORT:0> */
+        String roomId = null;
+        CallCompositeParticipantRole roomRole = null;
+        /* </ROOMS_SUPPORT:0> */
         final CallType callType;
 
         final CallCompositeJoinLocator locator = remoteOptions.getLocator();
         if (locator instanceof CallCompositeGroupCallLocator) {
             callType = CallType.GROUP_CALL;
             groupId = ((CallCompositeGroupCallLocator) locator).getGroupId();
-        } else {
+        } else if (locator instanceof CallCompositeTeamsMeetingLinkLocator) {
             callType = CallType.TEAMS_MEETING;
             meetingLink = ((CallCompositeTeamsMeetingLinkLocator) locator).getMeetingLink();
+            /* <ROOMS_SUPPORT:0> */
+        } else if (locator instanceof CallCompositeRoomLocator) {
+            callType = CallType.ROOMS_CALL;
+            final CallCompositeRoomLocator roomLocator = (CallCompositeRoomLocator) locator;
+            roomId = roomLocator.getRoomId();
+            /* </ROOMS_SUPPORT:0> */
+        } else {
```

Comment:
this is a new codepath, what's the expected behavior on UI if somehow users end up in here?
before we would use else statement for TeamsLocator

Reply:
It should never happen based on the public API. Added CallCompositeException out of precaution.